sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+            configured_par10: The PAR10 result of the configured solver
+            default_par10: The PAR10 result of the default solver
+        """
+        self.configured_PAR10 = configured_par10
+        self.default_PAR10 = default_par10
+
+
+class ValResults:
+    """Class that stores validation information and results."""
+    def __init__(self: ValResults, solver: str, configuration: str, instance_set: str,
+                 results: list[list[str, Status, float, float]]) -> None:
+        """Initalize ValResults.
+
+        Args:
+            solver: The name of the solver
+            configuration: The configuration being used
+            instance_set: The set of instances
+            results: Validation results in the format:
+                [["instance", "status", "quality", "runtime"]]
+        """
+        self.solver = solver
+        self.configuration = configuration
+        self.instance_set = instance_set
+        self.result_header = ["instance", "status", "quality", "runtime"]
+        self.result_vals = results
+
+
+class ConfigurationResults:
+    """Class that aggregates configuration results."""
+    def __init__(self: ConfigurationResults, performance: ConfigurationPerformance,
+                 res_def: ValResults, res_conf: ValResults) -> None:
+        """Initalize ConfigurationResults.
+
+        Args:
+            performance: The performance of the default and configured solver
+            res_def: The default results
+            res_conf: The configured results
+        """
+        self.performance = performance
+        self.configured_results = res_conf
+        self.default_results = res_def
+
+
+class ConfigurationOutput:
+    """Class that collects configuration data and outputs it a json format."""
+    solver: Solver = None
+    configurator: Configurator = None
+    best_config: str = None
+    training: ConfigurationResults = None
+    test: ConfigurationResults = None
+
+    instance_set_test: InstanceSet = None
+
+    def __init__(self: ConfigurationOutput, path: Path, solver_name: str,
+                 configurator: Configurator, penalty_multiplier: int,
+                 output: Path = None) -> None:
+        """Initialize Configurator Output class.
+
+        Args:
+            path: Path to configuration directory
+            solver_name: Name of the solver that was used
+            configurator: The configurator that was used
+            penalty_multiplier: penalty multiplier that is applied to the par performance
+            output: Path to the output directory
+        """
+        self.solver = resolve_object_name(solver_name,
+                                          gv.solver_nickname_mapping,
+                                          gv.solver_dir, Solver)
+        self.configurator = configurator
+        self.penalty_multiplier = penalty_multiplier
+        self.directory = path
+
+        if output is None:
+            output = path / "Analysis" / "configuration.json"
```

Comment:
Hard coded string, avoid

Reply:
Should I add it to global_variables?

Reply:
Maybe it can be refactored out in a later stage, but I'd suggest just saying something like Path("configuration.json")

Reply:
I'm not sure what you mean. What exactly should I change?

Reply:
/ "Analysis" / should not be in there hard coded

Reply:
But I'm not sure what the solution is
As the path variable is dependent on the configurator, we can't make a global variable which points to the path / "Analysis" folder
Should I then just make a variable that holds "Analysis"?